pkg/splunk/enterprise/types.go

```diff
@@ -57,6 +57,18 @@ const (
 	maxRunTimeBeforeAttemptingYield = 90
 )
 
+type globalResourceTracker struct {
+	storage *storageTracker
+}
+
+type storageTracker struct {
+	// represents the available disk space on operator pod
+	availableDiskSpace uint64
+
+	// mutex to serialize the access
+	mutex sync.Mutex
```

Comment:
IMHO may be the better place for this mutex is in `globalResourceTracker`, that way in the future you don't have to create a separate mutex for each resource type to track.

Reply:
From a long-term perspective, it is more logical to have a separate lock for each resource type(ex. CPU, memory, n/wconnections, etc) as new trackers are being added. That way each resource tracker can be independently handled/accessed/modified.